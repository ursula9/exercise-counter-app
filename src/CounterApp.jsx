import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    // function handleAdd( event ) { 
    //     console.log(event) 
    // }

    // function handleAdd( event, newValue ) { 
    //     console.log(newValue) 
    // }

    console.log('render');

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => { 
    //const handleAdd = ( event, newValue ) => { 
        //console.log(newValue) 
        //setCounter( counter + 1 );
        //setCounter(1000);
        setCounter( (c) => c + 1 );
    };

    const handleMinus = () => {
        setCounter( (c) => c - 1);
    }

    const handleReset = () => {
        setCounter(value);
    }



    return(
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ ( event ) => handleAdd(event, 'hola') }>  +1 </button>
            <button onClick={ ( event ) => handleMinus(event, 'minus')}> -1 </button>
            <button arial-label="btn-reset" onClick={ ( event ) => handleReset(event, 'Reseteado') }> Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}
