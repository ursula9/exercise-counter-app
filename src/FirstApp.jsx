import PropTypes from 'prop-types';
 
// export const FirstApp = () => {
//     return(
//         <div>
//             <h2>Monky Bebé</h2>
//             <p>guapito</p>
//         </div>
//      );
// }

// import { Fragment } from "react";

// export const FirstApp = () => {
//     return(
//         <Fragment>
//             <h2>Monky Bebé</h2>
//             <p>guapito</p>
//         </Fragment>
//      );
// }

//const newMessage = [1,2,3,4,5,6,7,8,9];
const newMessage = {
    message: 'holi',
    title: 'Ursula'
};

const mensajeFuncion = () => {
    return (
        'monkitos como tu'
    );
};

// export const FirstApp = ( props ) => {

 export const FirstApp = ( {
     title, 
     subTitle = 'La tierra explota!',
     name
 } ) => {
// console.log( props );

    if ( !title ) {
        throw new Error('El title no existe');
    }

    return(
        <>
            {/* <h1>{ mensajeFuncion() }</h1>  */}
            {/* <h1> { title } </h1> */}
            <h1 data-testid="test-title">{ title }</h1>
            <h2>Monky Bebé { 1+1 }  </h2>
            <p>guapito </p>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
            {/* <code>{ JSON.stringify( newMessage )}</code> */}
        </>
     );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    title: 'No hay título',
    subTitle: 'No hay subtítulo',
    name: 'Ursula Palacios'
}