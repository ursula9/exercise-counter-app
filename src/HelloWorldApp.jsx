
// function App () {
//     // document.createElement ...
//     return (<h1>Hola mundo !!!</h1>);
// }

// export default App;

// export const HelloWorldApp = () => {
//     return (
//         <div>HelloWorldApp</div>
//     );
// }

export const HelloWorldApp = () => <h1>Arrow Function</h1>