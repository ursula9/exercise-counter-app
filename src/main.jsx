import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';

//import App from './HelloWorldApp';

import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <HelloWorldApp/>
        {/* <FirstApp title="chopp" subTitle={ 123 } /> */}
        {/* <FirstApp title="hola, soy chopper"/> */}
        <CounterApp value={20}/>
    </React.StrictMode>
)