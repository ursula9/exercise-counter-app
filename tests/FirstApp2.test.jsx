import { getRoles, render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe( 'Pruebas en <FirstApp />', () => {

    const title    = 'Hola, soy Goku';
    const subTitle = 'Soy un subtÃ­tulo';

    test('debe de hacer match cone l snaptshot' , () => {

        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();

    });

    test( 'debe de mostrar el mensaje "Hola, soy Goku"', () => {

        
        render( <FirstApp title={ title } /> );
        //screen.debug();
        expect(  screen.getByText(title)).toBeTruthy();

    });

    test('Debe de mostrar el titulo en un h1', () => {

        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );

    });

    test('debe de mostrar el subtitulo enviado por props', () => {

        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />
        );
        expect(  screen.getAllByText(subTitle).length).toBe(2);


    });

});