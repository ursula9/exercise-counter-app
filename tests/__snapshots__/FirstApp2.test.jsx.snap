// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match cone l snaptshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <h2>
    Monky Bebé 
    2
      
  </h2>
  <p>
    guapito 
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Ursula Palacios
  </p>
</div>
`;
